packages/rum-core/src/domain/tracing/tracer.ts

```diff
@@ -113,7 +144,27 @@ function getCrypto() {
  * When trace is not sampled, set priority to '0' instead of not adding the tracing headers
  * to prepare the implementation for sampling delegation.
  */
-function makeTracingHeaders(traceId: TraceIdentifier, spanId: TraceIdentifier, traceSampled: boolean): TracingHeaders {
+function makeTracingHeaders(
+  traceId: TraceIdentifier,
+  spanId: TraceIdentifier,
+  traceSampled: boolean,
+  requestedHeaders: TracingHeadersType[]
+): TracingHeaders {
+  return requestedHeaders.reduce(
+    (headers, request) =>
+      assign(
+        headers,
+        makeTracingHeadersFor[request] ? makeTracingHeadersFor[request](traceId, spanId, traceSampled) : {}
+      ),
+    {} as TracingHeaders
+  )
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏here too I think the `reduce` complicates things a bit, but that's just my opinion, feel free to ignore